src/modules/hip/kernel/non_silent_region_detection.hpp

```diff
@@ -0,0 +1,401 @@
+    uint srcLength = srcLengthTensor[id_z];
+    float cutOffMag = cutOffMagPtr[id_z];
+
+    __shared__ int beginResult;
+    __shared__ int endResult;
+    beginResult = srcLength;
+    endResult = 0;
+    __syncthreads();
+
+    int beginIdx = srcLength;
+    int endIdx = 0;
+    for (int i = id_x; i < srcLength; i += hipBlockDim_x)
+    {
+        uint srcIdx = id_z * nStride + i;
+        if (srcPtr[srcIdx] >= cutOffMag)
+        {
+            beginIdx = i;
+            atomicMin(&beginResult, beginIdx);
+            if(beginResult != srcLength)
+                break;
+        }
+    }
+    for (int i = id_x; i < srcLength; i += hipBlockDim_x)
+    {
+        uint srcIdx = id_z * nStride + srcLength - 1 - i;
+        if (srcPtr[srcIdx] >= cutOffMag)
+        {
+            endIdx = srcLength - 1 - i;
+            atomicMax(&endResult, endIdx);
+            if(endResult != 0)
+                break;
+        }
+    }
+    if(hipThreadIdx_x == 0)
+    {
+        if(beginResult == srcLength || endResult == 0)
+        {
+            beginTensor[id_z] = 0;
+            lengthTensor[id_z] = 0;
+        }
+        else
+        {
+            int detectBegin = beginResult;
+            int detectEnd = endResult - beginResult + 1;
+            if(detectBegin != 0 && detectEnd != 0)
+            {
+                int newBegin = max(detectBegin - (windowLength - 1), 0);
+                detectEnd += detectBegin - newBegin;
+                detectBegin = newBegin;
+            }
+            beginTensor[id_z] = detectBegin;
+            lengthTensor[id_z] = detectEnd;
+        }
+    }
+}
+
+// -------------------- Set 4 -  host helpers for kernel executor --------------------
+
+int prev_pow2(int n)
+{
+    int pow2 = 1;
+    while (n - pow2 > pow2)
+        pow2 += pow2;
+
+    return pow2;
+}
+
+int next_pow2(int n)
+{
+    int pow2 = 1;
+    while (n > pow2)
+        pow2 += pow2;
+
+    return pow2;
+}
+
+// -------------------- Set 5 - non silent region kernels executor --------------------
+
+RppStatus hip_exec_non_silent_region_detection_tensor(Rpp32f *srcPtr,
+                                                      RpptDescPtr srcDescPtr,
+                                                      Rpp32s *srcLengthTensor,
+                                                      Rpp32s *detectedIndexTensor,
+                                                      Rpp32s *detectionLengthTensor,
+                                                      Rpp32f cutOffDB,
+                                                      Rpp32s windowLength,
+                                                      Rpp32f referencePower,
+                                                      Rpp32s resetInterval,
+                                                      rpp::Handle& handle)
+{
+    // allocate temporary memory for MMS Array
+    Rpp32f *mmsArr;
+    hipMalloc(&(mmsArr), srcDescPtr->n * srcDescPtr->strides.nStride * sizeof(Rpp32f));
```

Comment:
This will require changes in HOST api too to pass this additional argument for temporary memory + changes in rocAL
Please let me know if we can proceed on this change